test/integration/scipy_test.py

```diff
@@ -0,0 +1,136 @@
+        ei = sys.exc_info()
+        try:
+            subprocess.check_call(["rm", "-rf", ENV_NAME])
+        except Exception as e:
+            print e
+        raise ei[0], ei[1], ei[2]
+
+SRC_DIR = ENV_NAME
+PYTHON_EXE = os.path.abspath(ENV_NAME + "/bin/python")
+CYTHON_DIR = os.path.abspath(os.path.join(SRC_DIR, "cython"))
+NUMPY_DIR = os.path.abspath(os.path.join(SRC_DIR, "numpy"))
+SCIPY_DIR = os.path.abspath(os.path.join(SRC_DIR, "scipy"))
+
+print_progress_header("Setting up Cython...")
+if not os.path.exists(CYTHON_DIR):
+
+    url = "https://github.com/cython/cython"
+    subprocess.check_call(["git", "clone", "--depth", "1", "--branch", "0.24", url], cwd=SRC_DIR)
+
+    if USE_CUSTOM_PATCHES:
+        PATCH_FILE = os.path.abspath(os.path.join(os.path.dirname(__file__), "Cython-0.24.patch"))
+        subprocess.check_call(["patch", "-p1", "--input=" + PATCH_FILE], cwd=CYTHON_DIR)
+        print ">>> Applied Cython patch"
+
+    try:
+        subprocess.check_call([PYTHON_EXE, "setup.py", "install"], cwd=CYTHON_DIR)
+        subprocess.check_call([PYTHON_EXE, "-c", "import Cython"], cwd=CYTHON_DIR)
+    except:
+        subprocess.check_call(["rm", "-rf", CYTHON_DIR])
+else:
+    print ">>> Cython already installed."
+
+env = os.environ
+CYTHON_BIN_DIR = os.path.abspath(os.path.join(ENV_NAME + "/bin"))
+env["PATH"] = CYTHON_BIN_DIR + ":" + env["PATH"]
+
+print_progress_header("Cloning up NumPy...")
+if not os.path.exists(NUMPY_DIR):
+    url = "https://github.com/numpy/numpy"
+    subprocess.check_call(["git", "clone", "--depth", "1", "--branch", "v1.11.0", url], cwd=SRC_DIR)
+else:
+    print ">>> NumPy already installed."
+
+try:
+    print_progress_header("Setting up NumPy...")
+    subprocess.check_call([PYTHON_EXE, "setup.py", "build"], cwd=NUMPY_DIR, env=env)
+
+    print_progress_header("Installing NumPy...")
+    subprocess.check_call([PYTHON_EXE, "setup.py", "install"], cwd=NUMPY_DIR, env=env)
+except:
+    subprocess.check_call(["rm", "-rf", NUMPY_DIR + "/build"])
+    subprocess.check_call(["rm", "-rf", NUMPY_DIR + "/dist"])
+
+    raise
+
+if not os.path.exists(SCIPY_DIR):
+    print_progress_header("Cloning up SciPy...")
+    url = "https://github.com/scipy/scipy"
+    subprocess.check_call(["git", "clone", "--depth", "1", "--branch", "v0.17.1", url], cwd=SRC_DIR)
+else:
+    print ">>> SciPy already installed."
+
+try:
+    print_progress_header("Setting up SciPy...")
+    subprocess.check_call([PYTHON_EXE, "setup.py", "build"], cwd=SCIPY_DIR, env=env)
+
+    print_progress_header("Installing SciPy...")
+    subprocess.check_call([PYTHON_EXE, "setup.py", "install"], cwd=SCIPY_DIR, env=env)
+except:
+    subprocess.check_call(["rm", "-rf", SCIPY_DIR + "/build"])
+    subprocess.check_call(["rm", "-rf", SCIPY_DIR + "/dist"])
+    raise
+
+scip_test = "import scipy; scipy.test(verbose=2)"
+
+print_progress_header("Running SciPy test suite...")
+subprocess.check_call([PYTHON_EXE, "-c", scip_test], cwd=CYTHON_DIR)
```

Comment:
would be nice if the test would compare the test result to a expected number of successes+failures like we do for numpy even though the test does not get run automatically (I did not run the test yet because of missing fortran compiler - but I suspect it prints some stats otherwise ignore my comment please)